pallets/compliance-manager/src/benchmarking.rs

```diff
@@ -0,0 +1,342 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use crate::*;
+use pallet_asset::SecurityToken;
+use pallet_balances as balances;
+use polymesh_common_utilities::{asset::AssetType, group::GroupTrait};
+use polymesh_primitives::{IdentityId, InvestorUid, TrustedFor, TrustedIssuer};
+
+use frame_benchmarking::{account, benchmarks};
+use frame_support::traits::Currency;
+use frame_system::RawOrigin;
+
+const SEED: u32 = 1;
+
+pub struct Account<T: Trait> {
+    pub account: T::AccountId,
+    pub origin: RawOrigin<T::AccountId>,
+    pub did: IdentityId,
+}
+
+fn uid_from_name_and_idx(name: &'static str, u: u32) -> InvestorUid {
+    InvestorUid::from((name, u).encode().as_slice())
+}
+
+fn make_account_without_did<T: Trait>(
+    name: &'static str,
+    u: u32,
+) -> (T::AccountId, RawOrigin<T::AccountId>) {
+    let account: T::AccountId = account(name, u, SEED);
+    let origin = RawOrigin::Signed(account.clone());
+    let _ = balances::Module::<T>::make_free_balance_be(&account, 1_000_000.into());
+    (account, origin)
+}
+
+fn make_account<T: Trait>(name: &'static str, u: u32) -> Account<T> {
```

Comment:
Let's keep these common functions at a commonplace. I have them defined in the Identity benchmarks and you can import them from there. Eventually, we can consider defining a new storage runtime for benchmarks as we have for tests.

Reply:
I agree. I have some work about a "common place" in another **local** branch in my next PR [MESH-1303](https://polymath.atlassian.net/browse/MESH-1303) ;)